# Generar un archivo tipo Notebook de Python que contenga el código fuente de:

 * Leer archivo fifa_eda.csv  

 * Elemento 1: Obtener matriz de correlación y dejarla como heat map 

 * Elemento 2: Gráfico de Correlación que responda – “Cuál es la relación entre la edad y el overall?” – explicar la salida 

 * Elemento 3: Generar un gráfico de barras por club que indique el número de jugadores 

 * Elemento 4: Hacer un gráfico multipanel que indique la relación altura (height) vs skill moves, siendo la variable del panel si es zurdo o derecho 

 * Elemento 5: Un gráfico que crea importante para mostrar a los potenciales cracks

In [1]:
# Lectura del archivo 
import pandas as pd
df = pd.read_csv('fifa_eda.csv')
df.head(5)

ID               Name  Age Nationality  Overall  Potential  \
0  158023           L. Messi   31   Argentina       94         94   
1   20801  Cristiano Ronaldo   33    Portugal       94         94   
2  190871          Neymar Jr   26      Brazil       92         93   
3  193080             De Gea   27       Spain       91         93   
4  192985       K. De Bruyne   27     Belgium       91         92   

                  Club     Value   Wage Preferred Foot  \
0         FC Barcelona  110500.0  565.0           Left   
1             Juventus   77000.0  405.0          Right   
2  Paris Saint-Germain  118500.0  290.0          Right   
3    Manchester United   72000.0  260.0          Right   
4      Manchester City  102000.0  355.0          Right   

   International Reputation  Skill Moves Position  Joined  \
0                       5.0          4.0       RF    2004   
1                       5.0          5.0       ST    2018   
2                       5.0          5.0       LW    2017   
3                       4.0          1.0       GK    2011   
4                       4.0          4.0      RCM    2015   

  Contract Valid Until    Height  Weight  Release Clause  
0           2021-01-01  5.583333   159.0        226500.0  
1           2022-01-01  6.166667   183.0        127100.0  
2           2022-01-01  5.750000   150.0        228100.0  
3           2020-01-01  6.333333   168.0        138600.0  
4           2023-01-01  5.916667   154.0        196400.0

In [7]:
# Elemento 1: Obtener la matriz de correlacion y dejarla como heatmap
#df.info()
# Como me marcar error al momento de usar la funcion CORR, solo tomo en cuenta loa valores numericos
import plotly.express as px
df_numericos = df.select_dtypes(include=['int64', 'float64'])
matriz_correlacion = df_numericos.corr()

In [5]:
#Muestro el resultado de la matiz
matriz_correlacion

ID       Age   Overall  Potential     Value  \
ID                        1.000000 -0.739208 -0.417025   0.047074 -0.139837   
Age                      -0.739208  1.000000  0.452350  -0.253312  0.078315   
Overall                  -0.417025  0.452350  1.000000   0.660939  0.631848   
Potential                 0.047074 -0.253312  0.660939   1.000000  0.579608   
Value                    -0.139837  0.078315  0.631848   0.579608  1.000000   
Wage                     -0.204610  0.141145  0.571926   0.486413  0.858086   
International Reputation -0.356191  0.253765  0.499491   0.372993  0.656158   
Skill Moves              -0.056914  0.027649  0.414463   0.354290  0.317246   
Joined                    0.206749 -0.202658 -0.169281  -0.047661 -0.115991   
Height                   -0.090090  0.082506  0.038527  -0.009791  0.002827   
Weight                   -0.191193  0.229940  0.154557  -0.006935  0.046702   
Release Clause           -0.121297  0.058672  0.597821   0.562346  0.973310   

                              Wage  International Reputation  Skill Moves  \
ID                       -0.204610                 -0.356191    -0.056914   
Age                       0.141145                  0.253765     0.027649   
Overall                   0.571926                  0.499491     0.414463   
Potential                 0.486413                  0.372993     0.354290   
Value                     0.858086                  0.656158     0.317246   
Wage                      1.000000                  0.668635     0.263205   
International Reputation  0.668635                  1.000000     0.208153   
Skill Moves               0.263205                  0.208153     1.000000   
Joined                   -0.142337                 -0.133009     0.020692   
Height                    0.019638                  0.034881    -0.422753   
Weight                    0.064764                  0.088340    -0.351209   
Release Clause            0.828161                  0.620863     0.297471   

                            Joined    Height    Weight  Release Clause  
ID                        0.206749 -0.090090 -0.191193       -0.121297  
Age                      -0.202658  0.082506  0.229940        0.058672  
Overall                  -0.169281  0.038527  0.154557        0.597821  
Potential                -0.047661 -0.009791 -0.006935        0.562346  
Value                    -0.115991  0.002827  0.046702        0.973310  
Wage                     -0.142337  0.019638  0.064764        0.828161  
International Reputation -0.133009  0.034881  0.088340        0.620863  
Skill Moves               0.020692 -0.422753 -0.351209        0.297471  
Joined                    1.000000  0.001188 -0.028274       -0.115374  
Height                    0.001188  1.000000  0.754678        0.001835  
Weight                   -0.028274  0.754678  1.000000        0.038103  
Release Clause           -0.115374  0.001835  0.038103        1.000000

In [10]:
# Creo el Heatmap
fig = px.imshow(matriz_correlacion, color_continuous_scale="RdBu", title="Matriz de correlación - Variables numéricas (FIFA)", aspect="auto")
fig.show()


# Explicacion de la salida:
  * La correlacion que hay entre la edad y el overall es de 0.452 que digamos que es media baja (mas cercana al cero).
  * Pero la grafica muestra que a partir de los 24 a;os los jugadores tienden a subir el overall

In [56]:
# Elemento 2: Gráfico de Correlación que responda – “Cuál es la relación entre la edad y el overall?” – explicar la salida 
df_age_overall = df[["Age", "Overall"]]
corr_age_overall = df_age_overall["Age"].corr(subset["Overall"])
print(f"La Correlación edad vs overall: {corr_age_overall:.3f}")
fig = px.scatter( df_age_overall, x="Age", y="Overall", title="Relación entre Edad y Overall")
fig.show()

La Correlación edad vs overall: 0.452


In [32]:
# Elemento 3: Generar un gráfico de barras por club que indique el número de jugadores 
df_jugadores_club = df.groupby(['Club']).size().rename('Numero').reset_index()

In [33]:
df_jugadores_club.head()

Club  Numero
0    SSV Jahn Regensburg      29
1  1. FC Heidenheim 1846      28
2   1. FC Kaiserslautern      26
3             1. FC Köln      28
4        1. FC Magdeburg      26

In [39]:
#En este caso inclui un sample 20 para que se pudiera ver mejor
fig = px.bar(df_jugadores_club.sample(20) , x = "Club", y="Numero", title="Numero de jugadores por club Sample 20", text="Numero", color="Club")
fig.show()

In [45]:
# Elemento 4: Hacer un gráfico multipanel que indique la relación altura (height) vs skill moves, siendo la variable del panel si es zurdo o derecho
fig = px.strip(df, x="Height", y="Skill Moves", facet_col="Preferred Foot", title="Relacion Altura vs Skill moves")
fig.show()

In [58]:
# Elemento 5: Un gráfico que crea importante para mostrar a los potenciales cracks
# Para esto seleecione el Desempe;o, el potencial y la edad
Seleccion = ["Overall", "Potential", "Age"]
cracks_potenciales = df[Seleccion].dropna()

fig = px.scatter(
    cracks_potenciales, x="Overall", y="Potential", color="Age", color_continuous_scale="Viridis", opacity=0.6, title="Mapa de cracks potenciales (Overall vs Potential)", labels={"Age": "Edad"})
fig.show()
